servers/javascript/src/tdk.ts

```diff
@@ -137,26 +138,31 @@ export class TDKImpl implements tdk.TDK, AsyncDisposable {
                 const coll = db.getCollection(normalizeCollectionID(update.collection));
                 const doc = await coll.getDocument(update.documentID);
                 if (!doc) throw new HTTPError(404, `No document "${update.documentID}"`);
+
+                const updatePath = (pathStr: string, value: cbl.CBLValue | undefined): void => {
+                    if (!KeyPathCache.path(pathStr).write(doc, value))
+                        throw new HTTPError(400, `Invalid path ${pathStr} in doc ${update.documentID}`);
+                };
+
                 switch (update.type) {
                     case 'UPDATE': {
                         if (update.updatedProperties) {
                             for (const props of update.updatedProperties) {
-                                for (const pathStr of Object.getOwnPropertyNames(props)) {
-                                    if (!this.#keyPaths.path(pathStr).write(doc, props[pathStr]))
-                                        throw new HTTPError(400, `Invalid path ${pathStr} in doc ${update.documentID}`);
-                                }
+                                for (const pathStr of Object.getOwnPropertyNames(props))
+                                    updatePath(pathStr, props[pathStr]);
                             }
                         }
                         if (update.removedProperties) {
                             for (const props of update.removedProperties) {
-                                for (const pathStr of props) {
-                                    if (!this.#keyPaths.path(pathStr).write(doc, undefined))
-                                        throw new HTTPError(400, `Invalid path ${pathStr} in doc ${update.documentID}`);
-                                }
+                                for (const pathStr of props)
+                                    updatePath(pathStr, undefined);
                             }
                         }
                         if (update.updatedBlobs) {
-                            throw new HTTPError(501, "updatedBlobs is not implemented yet"); //TODO
+                            for (const pathStr of Object.getOwnPropertyNames(update.updatedBlobs)) {
+                                const blob = await this.#downloadBlob(update.updatedBlobs[pathStr]);
```

Comment:
I think this may need to be combined with the blob base URL?

Reply:
What's the blob base URL?